File: Special Array With X Elements Greater Than or Equal X.py
class Solution:
    def specialArray(self, nums: List[int]) -> int:
        nums.sort()
        n = len(nums)

        for i in range(n):
            if i != 0 and nums[i] == nums[i-1]:
                continue
            if n-i < nums[i] and (nums[i-1] < n-i or i == 0):
                return n-i
            if nums[i] == n-i:
                return n-i
        return -1
    

File: Special Array With X Elements Greater Than or Equal X.cpp
class Solution {
public:
    int specialArray(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int n = nums.size();

        for (int i = 0; i < n; i++) {
            if (i != 0 && nums[i] == nums[i-1]) continue;
            if ((n-i < nums[i] && (i == 0 || nums[i-1] < n-i)) || nums[i] == n-i ) return n-i;
        }
        return -1;
    }
};
